core-bundle/src/Mailer/ContaoMailer.php

```diff
@@ -115,4 +104,34 @@ private function setFrom(Email $message): void
             $message->sender($from);
         }
     }
+
+    /**
+     * Copy of AbstractFragmentController::getPageModel.
+     */
+    private function getPageModel(): ?PageModel
+    {
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (null === $request || !$request->attributes->has('pageModel')) {
+            return null;
+        }
+
+        $pageModel = $request->attributes->get('pageModel');
+
+        if ($pageModel instanceof PageModel) {
+            return $pageModel;
+        }
+
+        if (
+            isset($GLOBALS['objPage'])
+            && $GLOBALS['objPage'] instanceof PageModel
+            && (int) $GLOBALS['objPage']->id === (int) $pageModel
+        ) {
+            return $GLOBALS['objPage'];
+        }
+
```

Comment:
```suggestion
```

I would remove this part and just use `findByPk()` in this case. If `$GLOBALS['objPage']` contains the page model, it will also be in the registry, so `findByPk()` does not add any overhead.

Reply:
But the same applies to `AbstractFragmentController::getPageModel` then, does it not? We discussed that this method should be an exact copy of that. So either we change it in both or neither.

Reply:
Then I am in favor of removing it in both methods, because even if `$GLOBALS['objPage']` is set and `->id === $pageModel` is true, the global object might be modified. Using `findByPk()` seems more reliable to me.

Reply:
But you always want the modified object.

Reply:
Okay, then never mind. It‘ Contao 4.13 anyway. 🤷‍♂️